src/ApiResource/ProblemError.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\ApiResource;
+
+use ApiPlatform\Metadata\ApiProperty;
+use ApiPlatform\Metadata\ErrorResource;
+
+#[ErrorResource(uriTemplate: '/problems/{status}', provider: 'api_platform.state_provider.default_problem_error')]
+class ProblemError extends \Exception
+{
+    public function __construct(
+        public readonly string $title,
+        public readonly string $detail,
+        #[ApiProperty(identifier: true)] public readonly int $status,
+        public readonly array $trace,
+        private ?string $instance = null,
+        private ?string $type = null
+    ) {
+        parent::__construct();
+    }
+
+    public static function createFromException(\Exception $exception, int $statusCode, ?string $instance = null, ?string $type = null): static
+    {
+        $problem = new static($exception->getMessage(), $exception->getMessage(), $statusCode, $exception->getTrace());
+        $problem->setType($type);
+        $problem->setInstance($instance);
+
+        return $problem;
+    }
+
+    public function getInstance(): ?string
+    {
+        return $this->instance;
+    }
+
+    public function setInstance(?string $instance): void
```

Comment:
Wouldn't it be better to make the object immutable either by never allowing it to be changed, or to use `withers` instead?